.github/workflows/e2e-init-chain.yml

```diff
@@ -0,0 +1,54 @@
+name: Build and Push E2E Init Chain Images
```

Comment:
Can you please add an explanation doc similar to this:
https://github.com/osmosis-labs/osmosis/blob/8ff4c7fe52657319f6b3c299c7002f7aa90d2d7b/.github/workflows/docker.yml#L1